### DISCUSSION TITLE
Postprocessors for values at single named node?

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#21047] [Q&A General] [answered]
I have an input mesh with some nodes I'd like to sample the solution at. The mesh contains separate nodesets for each of those nodes (each nodeset contains exactly one node). 

I thought I could do a boundary-restricted Postprocessor but `construct_side_list_from_node_list` doesn't seem to work for single-node nodesets. The MOOSE Postprocessors that operate at node take the node id, which I don't necessarily know. I'd like to use the nodeset name instead.

Any ideas how to handle this? Could I make a new Postprocessor that accepts a nodeset name? Is there an example of that anywhere?

**Comment by GiudGiud** [comment, marked as answer]
Can you try the point value sampler?

**Reply by jessecarterMOOSE** [reply]
That takes the coordinates of the point which, like the node id, I may not know. I want to use the name of the nodeset.

**Reply by jessecarterMOOSE** [reply]
To give you a bit more information, what I'm doing is generating arbitrary meshes with another program and reading them into MOOSE. The meshes are coming in with the nodes labelled ("node1", "node2", etc) where I'd like to evaluate the solution. The labels are actually on the single-node-containing nodesets.

**Reply by GiudGiud** [reply]
oh ok my bad. i think NodalValueSampler will work, with a boundary parameter that is the nodeset
https://mooseframework.inl.gov/source/vectorpostprocessors/NodalValueSampler.html

**Reply by GiudGiud** [reply]
if it doesnt work we can try this one too
https://mooseframework.inl.gov/source/vectorpostprocessors/SideValueSampler.html

but side should be a sideset not a nodeset. Let's see

**Reply by jessecarterMOOSE** [reply]
OK, those work. Also AverageNodalVariableValue with boundary=\<nodeset name\> works. Guess I was confused between nodeset, sideset, and boundary. Thanks!
